src/hotspot/share/interpreter/rewriter.hpp

```diff
@@ -68,24 +70,14 @@ class Rewriter: public StackObj {
     _resolved_references_map.trunc_to(0);
     _invokedynamic_references_map.trunc_to(0);
     _resolved_reference_limit = -1;
-    _first_iteration_cp_cache_limit = -1;
   }
 
-  int _first_iteration_cp_cache_limit;
   void record_map_limits() {
     // Record initial size of the two arrays generated for the CP cache
     // relative to walking the constant pool.
-    _first_iteration_cp_cache_limit = _cp_cache_map.length();
     _resolved_reference_limit = _resolved_references_map.length();
   }
 
-  int cp_cache_delta() {
-    // How many cp cache entries were added since recording map limits after
-    // cp cache initialization?
-    assert(_first_iteration_cp_cache_limit != -1, "only valid after first iteration");
-    return _cp_cache_map.length() - _first_iteration_cp_cache_limit;
-  }
-
```

Comment:
I'm so pleased this special case went away.  We still need to add entries for invokespecial of JVM_CONSTANT_InterfaceMethodref but we don't need to keep track of where the original entries ended anymore with your changes.